### DISCUSSION TITLE
Atmospheric correction without science+standard

### DISCUSSION THREAD

**Discussion by rjs3273** [cylammarco/ASPIRED#115] [General]
The docs say current dev version of the atmospheric extinction correction only works when provided a science and a standard together. I looked through the code and I think the algorithm is currently moving the science target to the same airmass as the standard observation. That is as opposed to the alternative strategy of moving both to airmass=1 or 0 or any known value. The standard is not modified, only the target. Is that right?

I presume the idea is that atmospheric corrections are rather poorly defined at the best of times so you want to keep the correction applied as small as possible. I can see sense in that.

What I am getting at is that if I want to apply atmospheric correction on a single spectrum, is there an easy workaround. If I load the same spectrum twice, tell ASPIRED they are science+standard and then apply_atmospheric_extinction_correction(standard_airmass=1), will it run correctly? I then have the target corrected to airmass=1 and I simply ignore and discard the standard. 

```
# Load the same single spec into science+standard?
target1D.from_twodspec(target2D, stype='science+standard')  
# Go through the wavelength and flux calibration as normal
target1D.set_atmospheric_extinction(location='orm')
target1D.apply_atmospheric_extinction_correction(standard_airmass=1)
```

Obviously it is only a bit of a workaround, but the code looks like it ought to work. Or is that too daft? Maybe better to just do the airmass correction myself until the ASPIRED function is finished. The manual does say it is work in progress.

Second question. Is apply_atmospheric_extinction_correction() supposed to be able to work on ADU before applying a flux calib? So far I have not been able to get it to work. It seems to act sensibly if I call it after apply_flux_calibration() but not if I call it before. Is that expected or should I open a new Issue for it?

By the way, there is a spelling typo in apply_atmospheric_extinction_correction() that actually has no effect on anything numerical: interpoalted_ext instead of interpolated_ext. It is only a temporary variable and never propagates anywhere so the spelloing is largely irrelevent.

**Comment by cylammarco** [comment]
It is supposed to work with the wavelength and flux calibrated spectra only, in both 0.4 and 0.5 series. Because: without wavelength calibration, flux cannot be calibrated. Without flux calibration, it doesn't make sense to me to calibrate higher-order effects in flux. However, if you wish to force it to work with pre-flux calibrated spectra, you can set 

> onedspec_object.science_flux_calibrated = True
> onedspec_object.standard_flux_calibrated = True
> onedspec_object.science_spectrum_list[0].flux = onedspec_object.science_spectrum_list[0].count
> onedspec_object.science_spectrum_list[0].flux_err = onedspec_object.science_spectrum_list[0].count_err
> onedspec_object.science_spectrum_list[0].sky = onedspec_object.science_spectrum_list[0].sky
> standard.....
> onedspec_object.apply_atmospheric_extinction_correction()

However, if you apply flux application after that, they will overwrite the atmospheric extinction correction...

**Comment by cylammarco** [comment]
typo fixed in b95b2e1 (dev-0.5.X)

**Comment by cylammarco** [comment]
Airmiss issue transferred to https://github.com/cylammarco/ASPIRED/issues/116
